code/game/mecha/equipment/tools/janitor_tools.dm

```diff
@@ -0,0 +1,363 @@
+			if(do_after(chassis.occupant, mop_speed, target = target))
+				for(var/turf/current_target_turf in view(1, target))
+					current_target_turf.cleaning_act(chassis.occupant, src, mop_speed, "mop", ".", skip_do_after = TRUE)
+				to_chat(chassis.occupant, "<span class='notice'>You mop \the [target].</span>")
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/post_clean(atom/target, mob/user)
+	var/turf/T = get_turf(target)
+	if(issimulatedturf(T))
+		reagents.reaction(T, REAGENT_TOUCH, 10)	// 10 is the multiplier for the reaction effect. 10 is needed to properly wet a floor.
+	reagents.remove_any(1)	// reaction() doesn't use up the reagents
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/can_clean()
+	if(reagents.has_reagent("water", 1) || reagents.has_reagent("cleaner", 1) || reagents.has_reagent("holywater", 1))
+		return TRUE
+	else
+		return FALSE
+
+// Auto-regeneration of water. Takes energy.
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/process()
+	if(reagents.total_volume < 1000)
+		reagents.add_reagent(refill_reagent, refill_rate)
+		chassis.use_power(refill_cost)
+		update_equip_info()
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/get_equip_info()
+	var/output = ..()
+	if(output)
+		return "[output] \[<a href='byond://?src=[UID()];toggle_mode=1'>Refill [refill_enabled? "Enabled" : "Disabled"]</a>\] \[[reagents.total_volume]\]"
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/Topic(href,href_list)
+	..()
+	var/datum/topic_input/afilter = new (href, href_list)
+	if(afilter.get("toggle_mode"))
+		refill_enabled = !refill_enabled
+		if(refill_enabled)
+			START_PROCESSING(SSobj, src)
+		else
+			STOP_PROCESSING(SSobj, src)
+		update_equip_info()
+
+#undef MOP_SOUND_CD
+
+// Light Replacer
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer
+	name = "\improper NT-12 illuminator"
+	desc = "A modified light replacer fit for an exosuit that zaps lights into place."
+	icon_state = "mecha_light_replacer"
+	equip_cooldown = 1.5 SECONDS
+	energy_drain = 100
+	range = MECHA_MELEE | 15
+
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer/emag_act(mob/user)
+	. = ..()
+	emagged = TRUE
+	to_chat(user, "<span class='notice'>You short out the safeties on [src].</span>")
+
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer/action(atom/target)
+	if(istype(target, /obj/machinery/light))
+		chassis.Beam(target, icon_state = "rped_upgrade", icon = 'icons/effects/effects.dmi', time = 5)
+		playsound(src, 'sound/items/pshoom.ogg', 40, 1)
+		ReplaceLight(target)
+
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer/proc/ReplaceLight(obj/machinery/light/target)
+	if(target.status != LIGHT_OK)
+		to_chat(chassis.occupant, "<span class='notice'>You replace [target.fitting] with [src].</span>")
+		target.status = LIGHT_OK
```

Comment:
Is this a copy-paste of how the base light replacement works? It might be better to put this fixing logic onto the light if so.

Reply:
It's very similar, as it doesn't have the shard refilling. I implemented the replacement code into the light object as a fix() proc.